test/sdn_migration/sdn_migration_test.go

```diff
@@ -0,0 +1,639 @@
+	gvr := schema.GroupVersionResource{
+		Group:    "config.openshift.io",
+		Version:  "v1",
+		Resource: resource,
+	}
+	return dynamicClient.Resource(gvr).Get(ctx, name, metav1.GetOptions{})
+}
+
+// checkMigrationStatus probes the status of the SDN-to-OVN migration
+func checkMigrationStatus(ctx context.Context, client *openshiftclient.Client) error {
+	var (
+		dynamicClient *dynamic.DynamicClient
+		err           error
+	)
+
+	if dynamicClient, err = getKubernetesDynamicClient(client); err != nil {
+		return &upgradeError{err: err}
+	}
+	for i := 1; i <= upgradeMaxAttempts; i++ {
+		// Get the current network configuration
+		networkConfig, err := getOpenshiftConfig(ctx, dynamicClient, "networks", "cluster")
+		if err != nil {
+			return fmt.Errorf("failed to get network configuration: %v", err)
+		}
+
+		// Extract the status conditions
+		status, found, err := unstructured.NestedMap(networkConfig.Object, "status")
+		if err != nil || !found {
+			return fmt.Errorf("failed to find status in network configuration: %v", err)
+		}
+
+		conditions, found, err := unstructured.NestedSlice(status, "conditions")
+		if err != nil || !found {
+			return fmt.Errorf("failed to find conditions in status: %v", err)
+		}
+
+		// Check for the migration condition
+		migrationInProgress := false
+		for _, cond := range conditions {
+			conditionMap, ok := cond.(map[string]interface{})
+			if !ok {
+				return fmt.Errorf("invalid type for condition entry")
+			}
+
+			conditionType, found, err := unstructured.NestedString(conditionMap, "type")
+			if err != nil || !found {
+				return fmt.Errorf("failed to find condition type: %v", err)
+			}
+
+			if conditionType == "NetworkTypeMigrationInProgress" {
+				status, found, err := unstructured.NestedString(conditionMap, "status")
+				if err != nil || !found {
+					return fmt.Errorf("failed to find condition status: %v", err)
+				}
+
+				reason, found, err := unstructured.NestedString(conditionMap, "reason")
+				if err != nil || !found {
+					return fmt.Errorf("failed to find condition reason: %v", err)
+				}
+
+				// Check if the migration is in progress
+				if status == "True" && reason == "NetworkTypeMigrationStarted" {
+					migrationInProgress = true
+					break
+				}
+
+				// Check if the migration is complete
+				if status == "False" && reason == "NetworkTypeMigrationCompleted" {
+					fmt.Println("Network migration completed successfully!")
+					return nil
+				}
+			}
+		}
+
+		if migrationInProgress {
+			fmt.Println("Network migration is still in progress...")
+		} else {
+			fmt.Println("Migration status is unknown or not in progress.")
+		}
+
+		time.Sleep(upgradeDelay * time.Second)
+	}
+
+	return fmt.Errorf("network migration is still in progress after %d attempts", upgradeMaxAttempts)
+
+}
+
+// checkUpgradeStatus probes the status of the cluster upgrade
+func checkUpgradeStatus(ctx context.Context, client *openshiftclient.Client, upgradeVersion string) error {
```

Comment:
If needed, we can extract some of that function out to deduplicate this

Reply:
The reason for not using https://github.com/openshift/osde2e-common/blob/b7010c1e369cc51ae0e27231271409b016b53b2d/pkg/openshift/osd/upgrade.go#L223 is because 
when I try to initiate an upgrade using the managedUpgrade operator  I would get this error "no version gate exists for 4.15"